upload/upload.go

```diff
@@ -138,7 +138,7 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 		}
 		fileExists, err := helpers.ListFiles(*config, listPrefix)
 		if err != nil {
-			log.Error("Couldn't get the file list ", err)
+			return fmt.Errorf("listing uploaded files: %s", err.Error())
```

Comment:
Just curious, in which condition will this error be triggered?

Reply:
Before the PR `sda-cli` would proceed with an upload regardless of an expired token. However, since the token was expired, the call to `helpers.ListFiles` would return an error because the user could not be authenticated. Then, the error would be `logged` instead of propagated, causing the `sda-cli` to continue and  `panic` somewhere down the road before the tool would finally exit.